src/auth/auth.service.ts

```diff
@@ -37,23 +42,37 @@ export class AuthService {
   }
 
   // Sign user in by email and password
-  async login(credentials: LoginUserDto): Promise<AccessToken> {
+  async login(response: Response, credentials: LoginUserDto): Promise<AccessToken> {
     const user = await this.validateUser(credentials);
     if(!user) {
       throw new UnauthorizedException();
     }
 
     // Add a username in the JWT token
-    const payload = {
+    const payload: JwtEncodedUserData = {
       sub: user['id'],
       username: user['username'],
       isAdmin: user['isAdmin']
     };
 
-    return {accessToken: await this.jwtService.signAsync(payload)};
+    const jwt = await this.jwtService.signAsync(payload);
+
+    response.cookie(AuthService.ACCESS_COOKIE_NAME, jwt, {
+      httpOnly: true,
+      maxAge: 6 * 60 * 60 * 1000 // 6 hours
+    })
+
+    return {accessToken: jwt};
+  }
+
+  async logout(response: Response) {
+    response.cookie(AuthService.ACCESS_COOKIE_NAME, "", {
+      httpOnly: true,
+      maxAge: 30 * 1000 // expire very soon, the cookie is empty anyways so no need to keep it
```

Comment:
also magic number, maybe in a const file :rocket:

Reply:
This is only for an expiring cookie, it won't be reused and is just a way to explicit safety in cookies